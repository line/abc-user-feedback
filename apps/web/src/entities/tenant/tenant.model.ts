/**
 * Copyright 2023 LINE Corporation
 *
 * LINE Corporation licenses this file to you under the Apache License,
 * version 2.0 (the "License"); you may not use this file except in compliance
 * with the License. You may obtain a copy of the License at:
 *
 *   https://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
 * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
 * License for the specific language governing permissions and limitations
 * under the License.
 */

import { create } from 'zustand';

import { client } from '@/shared';

import type { Tenant } from './tenant.type';

type State = { tenant: Tenant | null };

type Action = {
  setTenant: (tenant: Tenant) => void;
  refetchTenant: () => Promise<void>;
};

export const useTenantStore = create<State & Action>((set) => ({
  tenant: null,
  setTenant: (tenant) => set({ tenant }),
  refetchTenant: async () => {
    const { data } = await client.get({ path: '/api/admin/tenants' });
    set({ tenant: data });
  },
}));
