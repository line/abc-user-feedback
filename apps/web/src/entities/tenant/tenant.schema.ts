/**
 * Copyright 2023 LINE Corporation
 *
 * LINE Corporation licenses this file to you under the Apache License,
 * version 2.0 (the "License"); you may not use this file except in compliance
 * with the License. You may obtain a copy of the License at:
 *
 *   https://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
 * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
 * License for the specific language governing permissions and limitations
 * under the License.
 */
import { z } from 'zod';

export const tenantSchema = z.object({
  id: z.number(),
  siteName: z.string(),
  description: z.string().nullable(),
  useEmail: z.boolean(),
  useOAuth: z.boolean(),
  isPrivate: z.boolean(),
  isRestrictDomain: z.boolean(),
  allowDomains: z
    .array(z.string().refine((v) => /[a-z]+\.[a-z]{2,3}/.test(v)))
    .nullable(),
  useEmailVerification: z.boolean(),
  oauthConfig: z
    .object({
      clientId: z.string(),
      emailKey: z.string(),
      scopeString: z.string(),
      clientSecret: z.string(),
      authCodeRequestURL: z.string(),
      accessTokenRequestURL: z.string(),
      userProfileRequestURL: z.string(),
    })
    .nullable(),
});

export const tenantInfoSchema = tenantSchema.pick({
  id: true,
  siteName: true,
  description: true,
});
export const authInfoScema = tenantSchema.pick({
  isPrivate: true,
  isRestrictDomain: true,
  allowDomains: true,
  useOAuth: true,
  useEmail: true,
  oauthConfig: true,
});
