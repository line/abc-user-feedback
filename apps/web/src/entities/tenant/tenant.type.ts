/**
 * Copyright 2023 LINE Corporation
 *
 * LINE Corporation licenses this file to you under the Apache License,
 * version 2.0 (the "License"); you may not use this file except in compliance
 * with the License. You may obtain a copy of the License at:
 *
 *   https://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
 * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
 * License for the specific language governing permissions and limitations
 * under the License.
 */

import { z } from 'zod';

const oauthConfigSchema = z.object({
  oauthUse: z.boolean(),
  clientId: z.string(),
  clientSecret: z.string(),
  authCodeRequestURL: z.string(),
  scopeString: z.string(),
  accessTokenRequestURL: z.string(),
  userProfileRequestURL: z.string(),
  emailKey: z.string(),
});

const tenantSchema = z.object({
  id: z.number(),
  siteName: z.string(),
  description: z.string().nullable(),
  useEmail: z.boolean(),
  useOAuth: z.boolean(),
  isPrivate: z.boolean(),
  isRestrictDomain: z.boolean(),
  allowDomains: z.array(z.string()),
  useEmailVerification: z.boolean(),
  oauthConfig: oauthConfigSchema.nullable(),
});

export type Tenant = z.infer<typeof tenantSchema>;

export type OAuthConfig = z.infer<typeof oauthConfigSchema>;
