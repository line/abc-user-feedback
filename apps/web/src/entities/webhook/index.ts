export * from './ui';
export * from './webhook.type';
