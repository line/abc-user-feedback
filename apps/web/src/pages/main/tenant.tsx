/**
 * Copyright 2025 LY Corporation
 *
 * LY Corporation licenses this file to you under the Apache License,
 * version 2.0 (the "License"); you may not use this file except in compliance
 * with the License. You may obtain a copy of the License at:
 *
 *   https://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
 * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
 * License for the specific language governing permissions and limitations
 * under the License.
 */
import type { GetStaticProps } from 'next';
import { useRouter } from 'next/router';
import { useTranslation } from 'next-i18next';
import { parseAsString, useQueryState } from 'nuqs';

import { Icon, Menu, MenuItem } from '@ufb/react';

import type { NextPageWithLayout } from '@/shared';
import SideMenuLayout from '@/shared/ui/side-menu-layout.ui';
import { Layout } from '@/widgets/layout';
import {
  LoginSetting,
  TenantInfoSetting,
  UserManagementSetting,
} from '@/widgets/setting-menu';

import serverSideTranslations from '@/server-side-translations';

const TenantPage: NextPageWithLayout = () => {
  const { t } = useTranslation();
  const router = useRouter();
  const [currentMenu] = useQueryState<string>(
    'menu',
    parseAsString.withDefault('tenant'),
  );

  return (
    <SideMenuLayout
      sideMenu={
        <Menu
          type="single"
          orientation="vertical"
          className="w-full p-0"
          value={currentMenu}
          onValueChange={(value) =>
            router.push({ pathname: router.pathname, query: { menu: value } })
          }
        >
          <MenuItem value="tenant">
            <Icon name="RiInformation2Line" />
            {t('tenant-setting-menu.tenant-info')}
          </MenuItem>
          <MenuItem value="login">
            <Icon name="RiUser2Line" />
            {t('tenant-setting-menu.sign-up-mgmt')}
          </MenuItem>
          <MenuItem value="user">
            <Icon name="RiShieldKeyholeLine" />
            {t('tenant-setting-menu.user-mgmt')}
          </MenuItem>
        </Menu>
      }
    >
      {currentMenu === 'tenant' && <TenantInfoSetting />}
      {currentMenu === 'login' && <LoginSetting />}
      {currentMenu === 'user' && <UserManagementSetting />}
    </SideMenuLayout>
  );
};

TenantPage.getLayout = (page) => {
  return <Layout title="Tenant">{page}</Layout>;
};

export const getStaticProps: GetStaticProps = async ({ locale }) => {
  return {
    props: {
      ...(await serverSideTranslations(locale)),
    },
  };
};

export default TenantPage;
