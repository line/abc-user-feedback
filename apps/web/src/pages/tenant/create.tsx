/**
 * Copyright 2023 LINE Corporation
 *
 * LINE Corporation licenses this file to you under the Apache License,
 * version 2.0 (the "License"); you may not use this file except in compliance
 * with the License. You may obtain a copy of the License at:
 *
 *   https://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
 * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
 * License for the specific language governing permissions and limitations
 * under the License.
 */
import type { GetStaticProps } from 'next';
import { useRouter } from 'next/router';
import { zodResolver } from '@hookform/resolvers/zod';
import { useTranslation } from 'next-i18next';
import { serverSideTranslations } from 'next-i18next/serverSideTranslations';
import { useForm } from 'react-hook-form';
import { z } from 'zod';

import { toast } from '@ufb/ui';

import AuthTemplate from '@/components/templates/AuthTemplate';
import { DEFAULT_LOCALE } from '@/constants/i18n';
import { Path } from '@/constants/path';
import { useTenant } from '@/contexts/tenant.context';
import { useOAIMutation } from '@/hooks';
import type { NextPageWithLayout } from '../_app';

interface IForm {
  siteName: string;
}

const schema: Zod.ZodType<IForm> = z.object({
  siteName: z.string(),
});

const defaultValues: IForm = {
  siteName: '',
};

const CreatePage: NextPageWithLayout = () => {
  const { t } = useTranslation();

  const router = useRouter();
  const { refetch } = useTenant();

  const { register, handleSubmit } = useForm({
    resolver: zodResolver(schema),
    defaultValues,
  });

  const { mutate, isPending } = useOAIMutation({
    method: 'post',
    path: '/api/admin/tenants',
    queryOptions: {
      async onSuccess() {
        toast.positive({ title: 'Success' });
        toast.positive({
          title: 'create Default Super User',
          description: 'email: user@feedback.com \n password: 12345678',
        });
        router.push(Path.SIGN_IN);
        refetch();
      },
      onError(error) {
        toast.negative({ title: error?.message ?? 'Error' });
      },
    },
  });
  const onSubmit = (data: IForm) => mutate(data);

  return (
    <AuthTemplate>
      <div className="flex h-screen w-screen items-center justify-center">
        <div className="w-full max-w-[400px] rounded border p-4 shadow">
          <form
            onSubmit={handleSubmit(onSubmit)}
            className="flex flex-col gap-4"
          >
            <h1 className="font-20-bold">{t('tenant.create.title')}</h1>
            <label>
              <span>{t('tenant.create.site-name')}</span>
              <input className="input" type="text" {...register('siteName')} />
            </label>

            <button
              className="btn btn-primary"
              type="submit"
              disabled={isPending}
            >
              {t('button.setting')}
            </button>
          </form>
        </div>
      </div>
    </AuthTemplate>
  );
};

export const getStaticProps: GetStaticProps = async ({ locale }) => {
  return {
    props: {
      ...(await serverSideTranslations(locale ?? DEFAULT_LOCALE)),
    },
  };
};

export default CreatePage;
