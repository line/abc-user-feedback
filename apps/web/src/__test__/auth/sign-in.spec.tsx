/**
 * Copyright 2023 LINE Corporation
 *
 * LINE Corporation licenses this file to you under the Apache License,
 * version 2.0 (the "License"); you may not use this file except in compliance
 * with the License. You may obtain a copy of the License at:
 *
 *   https://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
 * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
 * License for the specific language governing permissions and limitations
 * under the License.
 */

import { faker } from '@faker-js/faker';

import type { Tenant } from '@/entities/tenant';
import { useTenantStore } from '@/entities/tenant';

import SignInPage from '@/pages/auth/sign-in';
import { render, screen } from '@/test-utils';

const DEFAULT_TENANT: Tenant = {
  id: 1,
  siteName: faker.string.sample(),
  description: null,
  allowDomains: [],
  isPrivate: false,
  useEmail: true,
  isRestrictDomain: false,
  oauthConfig: null,
  useEmailVerification: true,
  useOAuth: true,
};

describe('Sign In Page', () => {
  test('should render all allow', () => {
    useTenantStore.setState({ tenant: { ...DEFAULT_TENANT } });
    const signInPage = SignInPage.getLayout?.(<SignInPage />);

    const { container } = render(<>{signInPage}</>);
    expect(container).toMatchSnapshot();

    const signInBtn = screen.queryByRole('button', { name: 'button.sign-in' });
    expect(signInBtn).toBeInTheDocument();

    const signUpBtn = screen.queryByRole('button', { name: 'button.sign-up' });
    expect(signUpBtn).toBeInTheDocument();

    const oauthBtn = screen.queryByRole('button', { name: /OAuth2.0/ });
    expect(oauthBtn).toBeInTheDocument();
  });
  test('should render when isPrivate is false', () => {
    useTenantStore.setState({ tenant: { ...DEFAULT_TENANT, isPrivate: true } });
    const signInPage = SignInPage.getLayout?.(<SignInPage />);

    const { container } = render(<>{signInPage}</>);
    expect(container).toMatchSnapshot();

    const signInBtn = screen.queryByRole('button', { name: 'button.sign-in' });
    expect(signInBtn).toBeInTheDocument();

    const signUpBtn = screen.queryByRole('button', { name: 'button.sign-up' });
    expect(signUpBtn).not.toBeInTheDocument();

    const oauthBtn = screen.queryByRole('button', { name: /OAuth2.0/ });
    expect(oauthBtn).toBeInTheDocument();
  });
  test('should render when useOAuth is false', () => {
    useTenantStore.setState({ tenant: { ...DEFAULT_TENANT, useOAuth: false } });
    const signInPage = SignInPage.getLayout?.(<SignInPage />);

    const { container } = render(<>{signInPage}</>);
    expect(container).toMatchSnapshot();

    const signInBtn = screen.queryByRole('button', { name: 'button.sign-in' });
    expect(signInBtn).toBeInTheDocument();

    const signUpBtn = screen.queryByRole('button', { name: 'button.sign-up' });
    expect(signUpBtn).toBeInTheDocument();

    const oauthBtn = screen.queryByRole('button', { name: /OAuth2.0/ });
    expect(oauthBtn).not.toBeInTheDocument();
  });
  test('should render when useEmail is false', () => {
    useTenantStore.setState({ tenant: { ...DEFAULT_TENANT, useEmail: false } });
    const signInPage = SignInPage.getLayout?.(<SignInPage />);

    const { container } = render(<>{signInPage}</>);
    expect(container).toMatchSnapshot();

    const signInBtn = screen.queryByRole('button', { name: 'button.sign-in' });
    expect(signInBtn).not.toBeInTheDocument();

    const signUpBtn = screen.queryByRole('button', { name: 'button.sign-up' });
    expect(signUpBtn).not.toBeInTheDocument();

    const oauthBtn = screen.queryByRole('button', { name: /OAuth2.0/ });
    expect(oauthBtn).toBeInTheDocument();
  });
});
