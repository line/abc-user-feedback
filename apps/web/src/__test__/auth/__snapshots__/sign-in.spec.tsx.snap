// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Sign In Page should render all allow 1`] = `
<div>
  <div
    class="min-w-[960px]"
  >
    <header
      class="relative flex h-[48px] items-center justify-between px-4"
    >
      <div
        class="flex flex-1 items-center gap-6"
      >
        <a
          class="flex cursor-pointer items-center gap-1"
          href="/auth/sign-in"
        >
          <img
            alt="logo"
            data-nimg="1"
            decoding="async"
            height="24"
            loading="lazy"
            src="/assets/images/logo.svg"
            style="color: transparent;"
            width="24"
          />
          <div
            class="inline-block h-[24px] w-[123px]"
            height="24"
            width="24"
          />
        </a>
        <div
          class="flex items-center gap-2"
        >
          <div
            class="flex items-center gap-2"
          >
            <div
              class="bg-fill-tertiary inline-flex rounded-sm p-1.5"
            >
              <div
                class="inline-block text-secondary"
                height="12"
                width="12"
              />
            </div>
            <span
              class="font-12-regular"
            >
              S_:GHQo.!/
            </span>
          </div>
        </div>
      </div>
      <div
        class="flex items-center gap-2 self-stretch"
      >
        <div
          class="relative"
          data-headlessui-state=""
        >
          <button
            aria-expanded="false"
            aria-haspopup="listbox"
            class="btn btn-sm btn-secondary min-w-0 px-2"
            data-headlessui-state=""
            id="headlessui-listbox-button-:r2:"
            type="button"
          >
            <div
              class="inline-block mr-1"
              height="16"
              width="16"
            />
            <span
              class="font-12-bold uppercase"
            />
          </button>
        </div>
        <button
          class="icon-btn icon-btn-sm icon-btn-secondary"
        >
          <div
            class="inline-block text-primary"
            height="24"
            width="24"
          />
        </button>
      </div>
    </header>
    <div
      class="bg-primary border-fill-secondary absolute left-1/2 top-1/2 min-w-[440px] -translate-x-1/2 -translate-y-1/2 rounded border p-8"
    >
      <div
        class="relative"
      >
        <div
          class="mb-8 flex flex-col items-center gap-1"
        >
          <img
            alt="logo"
            data-nimg="1"
            decoding="async"
            height="50"
            loading="lazy"
            src="/assets/images/logo-horizontal.svg"
            style="color: transparent;"
            width="124.5"
          />
          <div
            class="font-14-regular w-full text-center"
          >
            S_:GHQo.!/
          </div>
        </div>
        <form>
          <div
            class="mb-6 flex flex-col gap-3"
          >
            <div
              class="flex flex-col gap-2"
            >
              <div
                class="relative"
              >
                <input
                  class="input  input-lg pl-10 pr-10 "
                  name="email"
                  placeholder="ID"
                  type="email"
                />
                <div
                  class="absolute-y-center absolute left-[14px] flex items-center gap-2"
                >
                  <div
                    class="inline-block text-tertiary"
                    height="20"
                    width="20"
                  />
                </div>
                <div
                  class="absolute-y-center absolute right-[14px] flex items-center gap-2"
                />
              </div>
            </div>
            <div
              class="flex flex-col gap-2"
            >
              <div
                class="relative"
              >
                <input
                  class="input  input-lg pl-10 pr-10 "
                  name="password"
                  placeholder="Password"
                  type="password"
                />
                <div
                  class="absolute-y-center absolute left-[14px] flex items-center gap-2"
                >
                  <div
                    class="inline-block text-tertiary"
                    height="20"
                    width="20"
                  />
                </div>
                <div
                  class="absolute-y-center absolute right-[14px] flex items-center gap-2"
                />
              </div>
            </div>
          </div>
          <div
            class="flex flex-col gap-1"
          >
            <button
              class="btn btn-lg btn-primary"
              disabled=""
              type="submit"
            >
              button.sign-in
            </button>
            <button
              class="btn btn-lg btn-secondary"
              type="button"
            >
              button.sign-up
            </button>
          </div>
        </form>
        <div
          class="my-1 flex flex-col gap-1"
        >
          <div
            class="relative my-5"
          >
            <span
              class="absolute-center text-secondary bg-primary absolute px-2 py-1"
            >
              OR
            </span>
            <hr />
          </div>
          <button
            class="btn btn-lg btn-blue"
            disabled=""
            type="button"
          >
            OAuth2.0 
            button.sign-in
          </button>
        </div>
        <div
          class="absolute -bottom-28 left-1/2 -translate-x-1/2"
        >
          <a
            class="text-blue-primary font-14-regular"
            href="/auth/reset-password"
          >
            auth.sign-in.reset-password
          </a>
        </div>
      </div>
    </div>
    <footer
      class="font-14-bold text-secondary absolute bottom-[7%] left-1/2 z-10 -translate-x-1/2"
    >
      Â© ABC Studio All rights reserved
    </footer>
  </div>
  <div
    style="position: fixed; z-index: 9999; top: 16px; left: 16px; right: 16px; bottom: 16px; pointer-events: none;"
  />
</div>
`;

exports[`Sign In Page should render when isPrivate is false 1`] = `
<div>
  <div
    class="min-w-[960px]"
  >
    <header
      class="relative flex h-[48px] items-center justify-between px-4"
    >
      <div
        class="flex flex-1 items-center gap-6"
      >
        <a
          class="flex cursor-pointer items-center gap-1"
          href="/auth/sign-in"
        >
          <img
            alt="logo"
            data-nimg="1"
            decoding="async"
            height="24"
            loading="lazy"
            src="/assets/images/logo.svg"
            style="color: transparent;"
            width="24"
          />
          <div
            class="inline-block h-[24px] w-[123px]"
            height="24"
            width="24"
          />
        </a>
        <div
          class="flex items-center gap-2"
        >
          <div
            class="flex items-center gap-2"
          >
            <div
              class="bg-fill-tertiary inline-flex rounded-sm p-1.5"
            >
              <div
                class="inline-block text-secondary"
                height="12"
                width="12"
              />
            </div>
            <span
              class="font-12-regular"
            >
              S_:GHQo.!/
            </span>
          </div>
        </div>
      </div>
      <div
        class="flex items-center gap-2 self-stretch"
      >
        <div
          class="relative"
          data-headlessui-state=""
        >
          <button
            aria-expanded="false"
            aria-haspopup="listbox"
            class="btn btn-sm btn-secondary min-w-0 px-2"
            data-headlessui-state=""
            id="headlessui-listbox-button-:r8:"
            type="button"
          >
            <div
              class="inline-block mr-1"
              height="16"
              width="16"
            />
            <span
              class="font-12-bold uppercase"
            />
          </button>
        </div>
        <button
          class="icon-btn icon-btn-sm icon-btn-secondary"
        >
          <div
            class="inline-block text-primary"
            height="24"
            width="24"
          />
        </button>
      </div>
    </header>
    <div
      class="bg-primary border-fill-secondary absolute left-1/2 top-1/2 min-w-[440px] -translate-x-1/2 -translate-y-1/2 rounded border p-8"
    >
      <div
        class="relative"
      >
        <div
          class="mb-8 flex flex-col items-center gap-1"
        >
          <img
            alt="logo"
            data-nimg="1"
            decoding="async"
            height="50"
            loading="lazy"
            src="/assets/images/logo-horizontal.svg"
            style="color: transparent;"
            width="124.5"
          />
          <div
            class="font-14-regular w-full text-center"
          >
            S_:GHQo.!/
          </div>
        </div>
        <form>
          <div
            class="mb-6 flex flex-col gap-3"
          >
            <div
              class="flex flex-col gap-2"
            >
              <div
                class="relative"
              >
                <input
                  class="input  input-lg pl-10 pr-10 "
                  name="email"
                  placeholder="ID"
                  type="email"
                />
                <div
                  class="absolute-y-center absolute left-[14px] flex items-center gap-2"
                >
                  <div
                    class="inline-block text-tertiary"
                    height="20"
                    width="20"
                  />
                </div>
                <div
                  class="absolute-y-center absolute right-[14px] flex items-center gap-2"
                />
              </div>
            </div>
            <div
              class="flex flex-col gap-2"
            >
              <div
                class="relative"
              >
                <input
                  class="input  input-lg pl-10 pr-10 "
                  name="password"
                  placeholder="Password"
                  type="password"
                />
                <div
                  class="absolute-y-center absolute left-[14px] flex items-center gap-2"
                >
                  <div
                    class="inline-block text-tertiary"
                    height="20"
                    width="20"
                  />
                </div>
                <div
                  class="absolute-y-center absolute right-[14px] flex items-center gap-2"
                />
              </div>
            </div>
          </div>
          <div
            class="flex flex-col gap-1"
          >
            <button
              class="btn btn-lg btn-primary"
              disabled=""
              type="submit"
            >
              button.sign-in
            </button>
          </div>
        </form>
        <div
          class="my-1 flex flex-col gap-1"
        >
          <div
            class="relative my-5"
          >
            <span
              class="absolute-center text-secondary bg-primary absolute px-2 py-1"
            >
              OR
            </span>
            <hr />
          </div>
          <button
            class="btn btn-lg btn-blue"
            disabled=""
            type="button"
          >
            OAuth2.0 
            button.sign-in
          </button>
        </div>
      </div>
    </div>
    <footer
      class="font-14-bold text-secondary absolute bottom-[7%] left-1/2 z-10 -translate-x-1/2"
    >
      Â© ABC Studio All rights reserved
    </footer>
  </div>
  <div
    style="position: fixed; z-index: 9999; top: 16px; left: 16px; right: 16px; bottom: 16px; pointer-events: none;"
  />
</div>
`;

exports[`Sign In Page should render when useEmail is false 1`] = `
<div>
  <div
    class="min-w-[960px]"
  >
    <header
      class="relative flex h-[48px] items-center justify-between px-4"
    >
      <div
        class="flex flex-1 items-center gap-6"
      >
        <a
          class="flex cursor-pointer items-center gap-1"
          href="/auth/sign-in"
        >
          <img
            alt="logo"
            data-nimg="1"
            decoding="async"
            height="24"
            loading="lazy"
            src="/assets/images/logo.svg"
            style="color: transparent;"
            width="24"
          />
          <div
            class="inline-block h-[24px] w-[123px]"
            height="24"
            width="24"
          />
        </a>
        <div
          class="flex items-center gap-2"
        >
          <div
            class="flex items-center gap-2"
          >
            <div
              class="bg-fill-tertiary inline-flex rounded-sm p-1.5"
            >
              <div
                class="inline-block text-secondary"
                height="12"
                width="12"
              />
            </div>
            <span
              class="font-12-regular"
            >
              S_:GHQo.!/
            </span>
          </div>
        </div>
      </div>
      <div
        class="flex items-center gap-2 self-stretch"
      >
        <div
          class="relative"
          data-headlessui-state=""
        >
          <button
            aria-expanded="false"
            aria-haspopup="listbox"
            class="btn btn-sm btn-secondary min-w-0 px-2"
            data-headlessui-state=""
            id="headlessui-listbox-button-:rk:"
            type="button"
          >
            <div
              class="inline-block mr-1"
              height="16"
              width="16"
            />
            <span
              class="font-12-bold uppercase"
            />
          </button>
        </div>
        <button
          class="icon-btn icon-btn-sm icon-btn-secondary"
        >
          <div
            class="inline-block text-primary"
            height="24"
            width="24"
          />
        </button>
      </div>
    </header>
    <div
      class="bg-primary border-fill-secondary absolute left-1/2 top-1/2 min-w-[440px] -translate-x-1/2 -translate-y-1/2 rounded border p-8"
    >
      <div
        class="relative"
      >
        <div
          class="mb-8 flex flex-col items-center gap-1"
        >
          <img
            alt="logo"
            data-nimg="1"
            decoding="async"
            height="50"
            loading="lazy"
            src="/assets/images/logo-horizontal.svg"
            style="color: transparent;"
            width="124.5"
          />
          <div
            class="font-14-regular w-full text-center"
          >
            S_:GHQo.!/
          </div>
        </div>
        <div
          class="my-1 flex flex-col gap-1"
        >
          <button
            class="btn btn-lg btn-blue"
            disabled=""
            type="button"
          >
            OAuth2.0 
            button.sign-in
          </button>
        </div>
      </div>
    </div>
    <footer
      class="font-14-bold text-secondary absolute bottom-[7%] left-1/2 z-10 -translate-x-1/2"
    >
      Â© ABC Studio All rights reserved
    </footer>
  </div>
  <div
    style="position: fixed; z-index: 9999; top: 16px; left: 16px; right: 16px; bottom: 16px; pointer-events: none;"
  />
</div>
`;

exports[`Sign In Page should render when useOAuth is false 1`] = `
<div>
  <div
    class="min-w-[960px]"
  >
    <header
      class="relative flex h-[48px] items-center justify-between px-4"
    >
      <div
        class="flex flex-1 items-center gap-6"
      >
        <a
          class="flex cursor-pointer items-center gap-1"
          href="/auth/sign-in"
        >
          <img
            alt="logo"
            data-nimg="1"
            decoding="async"
            height="24"
            loading="lazy"
            src="/assets/images/logo.svg"
            style="color: transparent;"
            width="24"
          />
          <div
            class="inline-block h-[24px] w-[123px]"
            height="24"
            width="24"
          />
        </a>
        <div
          class="flex items-center gap-2"
        >
          <div
            class="flex items-center gap-2"
          >
            <div
              class="bg-fill-tertiary inline-flex rounded-sm p-1.5"
            >
              <div
                class="inline-block text-secondary"
                height="12"
                width="12"
              />
            </div>
            <span
              class="font-12-regular"
            >
              S_:GHQo.!/
            </span>
          </div>
        </div>
      </div>
      <div
        class="flex items-center gap-2 self-stretch"
      >
        <div
          class="relative"
          data-headlessui-state=""
        >
          <button
            aria-expanded="false"
            aria-haspopup="listbox"
            class="btn btn-sm btn-secondary min-w-0 px-2"
            data-headlessui-state=""
            id="headlessui-listbox-button-:re:"
            type="button"
          >
            <div
              class="inline-block mr-1"
              height="16"
              width="16"
            />
            <span
              class="font-12-bold uppercase"
            />
          </button>
        </div>
        <button
          class="icon-btn icon-btn-sm icon-btn-secondary"
        >
          <div
            class="inline-block text-primary"
            height="24"
            width="24"
          />
        </button>
      </div>
    </header>
    <div
      class="bg-primary border-fill-secondary absolute left-1/2 top-1/2 min-w-[440px] -translate-x-1/2 -translate-y-1/2 rounded border p-8"
    >
      <div
        class="relative"
      >
        <div
          class="mb-8 flex flex-col items-center gap-1"
        >
          <img
            alt="logo"
            data-nimg="1"
            decoding="async"
            height="50"
            loading="lazy"
            src="/assets/images/logo-horizontal.svg"
            style="color: transparent;"
            width="124.5"
          />
          <div
            class="font-14-regular w-full text-center"
          >
            S_:GHQo.!/
          </div>
        </div>
        <form>
          <div
            class="mb-6 flex flex-col gap-3"
          >
            <div
              class="flex flex-col gap-2"
            >
              <div
                class="relative"
              >
                <input
                  class="input  input-lg pl-10 pr-10 "
                  name="email"
                  placeholder="ID"
                  type="email"
                />
                <div
                  class="absolute-y-center absolute left-[14px] flex items-center gap-2"
                >
                  <div
                    class="inline-block text-tertiary"
                    height="20"
                    width="20"
                  />
                </div>
                <div
                  class="absolute-y-center absolute right-[14px] flex items-center gap-2"
                />
              </div>
            </div>
            <div
              class="flex flex-col gap-2"
            >
              <div
                class="relative"
              >
                <input
                  class="input  input-lg pl-10 pr-10 "
                  name="password"
                  placeholder="Password"
                  type="password"
                />
                <div
                  class="absolute-y-center absolute left-[14px] flex items-center gap-2"
                >
                  <div
                    class="inline-block text-tertiary"
                    height="20"
                    width="20"
                  />
                </div>
                <div
                  class="absolute-y-center absolute right-[14px] flex items-center gap-2"
                />
              </div>
            </div>
          </div>
          <div
            class="flex flex-col gap-1"
          >
            <button
              class="btn btn-lg btn-primary"
              disabled=""
              type="submit"
            >
              button.sign-in
            </button>
            <button
              class="btn btn-lg btn-secondary"
              type="button"
            >
              button.sign-up
            </button>
          </div>
        </form>
        <div
          class="my-1 flex flex-col gap-1"
        />
        <div
          class="absolute -bottom-28 left-1/2 -translate-x-1/2"
        >
          <a
            class="text-blue-primary font-14-regular"
            href="/auth/reset-password"
          >
            auth.sign-in.reset-password
          </a>
        </div>
      </div>
    </div>
    <footer
      class="font-14-bold text-secondary absolute bottom-[7%] left-1/2 z-10 -translate-x-1/2"
    >
      Â© ABC Studio All rights reserved
    </footer>
  </div>
  <div
    style="position: fixed; z-index: 9999; top: 16px; left: 16px; right: 16px; bottom: 16px; pointer-events: none;"
  />
</div>
`;
